drivers/adc/adm1177/adm1177.c

```diff
@@ -0,0 +1,316 @@
+ * 			will be set to 0.
+ *
+ * @return 0 in case of success, negative error code otherwise.
+******************************************************************************/
+int adm1177_write(struct adm1177_dev *device, uint8_t cmd, uint8_t reg_address)
+{
+	if(reg_address & ADM1177_EXT_CMD_MASK) {
+		uint8_t data[2] = {reg_address, cmd};
+
+		return no_os_i2c_write(device->i2c_desc, data, 2, 1);
+	} else {
+		if(cmd & (ADM1177_CMD_I_CONT | ADM1177_CMD_I_ONCE)) {
+			device->last_command = ADM1177_CURRENT_EN;
+			return no_os_i2c_write(device->i2c_desc, &cmd, 1, 1);
+		}
+		if(cmd & (ADM1177_CMD_V_CONT | ADM1177_CMD_V_ONCE)) {
+			device->last_command = ADM1177_VOLTAGE_EN;
+			return no_os_i2c_write(device->i2c_desc, &cmd, 1, 1);
+		}
+		if(cmd & (ADM1177_CMD_I_CONT | ADM1177_CMD_V_CONT) |
+		    cmd & (ADM1177_CMD_I_ONCE | ADM1177_CMD_V_ONCE)) {
+			device->last_command = ADM1177_VOLTAGE_AND_CURRENT_EN;
+			return no_os_i2c_write(device->i2c_desc, &cmd, 1, 1);
+		}
+		if(cmd & ADM1177_CMD_STATUS_RD) {
+			device->last_command = ADM1177_STATUS_EN;
+			return no_os_i2c_write(device->i2c_desc, &cmd, 1, 1);
+		}
+		if(cmd & ADM1177_CMD_VRANGE) {
+			device->vrange = ADM1177_VRANGE_HIGH;
+		} else {
+			device->vrange = ADM1177_VRANGE_LOW;
+		}
+		return -EINVAL;
+	}
+}
+
+/**************************************************************************//**
+ * @brief - Reads the status byte from the ADC if requested and transmits it
+ * 	    through the status_byte pointer.
+ *
+ * @param device - The device structure.
+ *
+ * @param status_byte - The pointer through which the status byte is
+ * 			transmitted.
+ *
+ * @return 0 in case of success, negative error code otherwise.
+******************************************************************************/
+int adm1177_read_status(struct adm1177_dev *device, uint8_t *status_byte)
+{
+	if(device->last_command == ADM1177_STATUS_EN) {
+		if(status_byte == NULL)
+			return -EINVAL;
+		int ret;
+
+		ret = no_os_i2c_read(device->i2c_desc, status_byte, 1, 1);
+		if(ret)
+			return ret;
+
+		return 0;
+	} else {
+		return -EINVAL;
+	}
+}
+
+/**************************************************************************//**
+ * @brief - Gets the voltage/current readback from the ADM1177s I2C ADC,
+ *	    if the device is initialized right and transmits the requested
+ *	    values.
+ *
+ * @param device - The device structure.
+ *
+ * @param conv_voltage - The variable through which the voltage read from the
+ * 			 ADC is received if requested.
+ * 			 If it is 0, it means it was not requested.
+ *
+ * @param conv_current - The variable through which the current read from the
+ * 			 ADC is received if requested.
+ * 			 If it is 0, it means it was not requested.
+ *
+ * @return 0 in case of success, negative error code otherwise.
+******************************************************************************/
+int adm1177_readback(struct adm1177_dev *device,
```

Comment:
i think `adm1177_read` is a better fit to complement existing API functions `adm1177_write` and `adm1177_read_status`

Reply:
Will change.